upup/pkg/fi/cloudup/spotinsttasks/elastigroup.go

```diff
@@ -637,44 +650,34 @@ func (_ *Elastigroup) create(cloud awsup.AWSCloud, a, e, changes *Elastigroup) e
 				}
 			}
 
-			// Load balancer.
+			// Load balancers.
 			{
-				if e.LoadBalancer != nil {
-					elb, err := cloud.FindELBByNameTag(fi.StringValue(e.LoadBalancer.Name))
+				if len(e.LoadBalancers) > 0 {
+					lbs, err := e.buildLoadBalancers(cloud)
 					if err != nil {
 						return err
 					}
-					if elb != nil {
-						lb := new(aws.LoadBalancer)
-						lb.SetName(elb.LoadBalancerName)
-						lb.SetType(fi.String("CLASSIC"))
-
-						cfg := new(aws.LoadBalancersConfig)
-						cfg.SetLoadBalancers([]*aws.LoadBalancer{lb})
 
-						group.Compute.LaunchSpecification.SetLoadBalancersConfig(cfg)
+					if group.Compute.LaunchSpecification.LoadBalancersConfig == nil {
+						group.Compute.LaunchSpecification.LoadBalancersConfig = new(aws.LoadBalancersConfig)
 					}
 
-					//TODO: Verify using NLB functionality
-					//TODO: Consider using DNSTarget Interface and adding .getLoadBalancerName() .getLoadBalancerArn
-					nlb, err := cloud.FindELBV2ByNameTag(fi.StringValue(e.LoadBalancer.Name))
-					if err != nil {
-						return err
-					}
-					if elb != nil && nlb != nil {
-						return fmt.Errorf("found both elb and nlb:")
-					}
-					if nlb != nil {
-						lb := new(aws.LoadBalancer)
-						lb.SetName(nlb.LoadBalancerName)
-						//lb.SetArn(nlb.LoadBalancerArn)
-						lb.SetType(fi.String("NETWORK"))
+					group.Compute.LaunchSpecification.LoadBalancersConfig.LoadBalancers =
+						append(group.Compute.LaunchSpecification.LoadBalancersConfig.LoadBalancers, lbs...)
+				}
+			}
 
-						cfg := new(aws.LoadBalancersConfig)
-						cfg.SetLoadBalancers([]*aws.LoadBalancer{lb})
+			// Target groups.
+			{
+				if len(e.TargetGroups) > 0 {
+					tgs := e.buildTargetGroups()
 
-						group.Compute.LaunchSpecification.SetLoadBalancersConfig(cfg)
+					if group.Compute.LaunchSpecification.LoadBalancersConfig == nil {
+						group.Compute.LaunchSpecification.LoadBalancersConfig = new(aws.LoadBalancersConfig)
 					}
+
+					group.Compute.LaunchSpecification.LoadBalancersConfig.LoadBalancers =
+						append(group.Compute.LaunchSpecification.LoadBalancersConfig.LoadBalancers, tgs...)
```

Comment:
```suggestion
					group.Compute.LaunchSpecification.LoadBalancersConfig.TargetGroups =
						append(group.Compute.LaunchSpecification.LoadBalancersConfig.TargetGroups, tgs...)
```

Reply:
Hi @rifelpet. Thanks for reviewing this! Please note that there is no `TargetGroups` field in the [LoadBalancersConfig](https://github.com/spotinst/spotinst-sdk-go/blob/v1.91.0/service/elastigroup/providers/aws/aws.go#L643-L648) object. It only takes a list of [LoadBalancer](https://github.com/spotinst/spotinst-sdk-go/blob/v1.91.0/service/elastigroup/providers/aws/aws.go#L650-L661) objects, and each of them defines the type of load balancer in its [Type](https://github.com/spotinst/spotinst-sdk-go/blob/v1.91.0/service/elastigroup/providers/aws/aws.go#L653) field.